File: Number of Students Unable to Eat Lunch.cpp
int sum(vector<int>& v) {
    int total = 0;
    for (int &i: v) total += i;
    return total;
}

class Solution {
public:
    int countStudents(vector<int>& students, vector<int>& sandwiches) {
        int n = sandwiches.size();
        int n1 = sum(students), n2 = sum(sandwiches);
        if (n1 == n2) return 0;

        int what = 0;
        if (n1-n2>0) what = 1;
        int count, other_count;
        if (what == 0) {
            count = n-n2;
            other_count = n1;
        } else {
            count=n2;
            other_count = n-n1;
        }

        for (int i=0; i<n; i++) {
            if (sandwiches[i]==what) {
                if (count > 0) count--;
                else return n-i;
            } else {
                if (other_count > 0) other_count--;
                else return n-i;
            }
        }
        return 0;
    }
};

File: Number of Students Unable to Eat Lunch.py
class Solution:
    def countStudents(self, students: List[int], sandwiches: List[int]) -> int:
        n = len(sandwiches)
        n1, n2 = sum(students), sum(sandwiches)
        if n1==n2:
            return 0

        what = int(n1-n2 > 0)
        if what == 0:
            count = n - n2
            other_count = n1
        else:
            count = n2
            other_count = n - n1

        for i, s in enumerate(sandwiches):
            if s == what:
                if count > 0:
                    count -= 1
                else:
                    return n-i
            else:
                if other_count > 0:
                    other_count -= 1
                else:
                    return n-i
        return 0
